### DISCUSSION TITLE
OLED display questions

### DISCUSSION THREAD

**Discussion by NorthBaandee** [timmbogner/Farm-Data-Relay-System#204] [Q&A] [answered]
I'm working on a wind turbine dump load governer that can be monitored by FDRS lora and controlled by a web page. I'm using a TTGO lora32 oled module. Rather than displaying debugging information I would like the oled to display battery voltage, charge current and dump load status ( ie diverted or not). Part of the design is a boost button that overrides the dump load diversion and forces the turbine to charge for a set time. This boost status needs to dominate the oled dispay, over riding everything else, so that such a situation is closely supervised. I'll provably also make it beep. If I set #define USE_OLED the debugging info continually wipes all the sensor info I write to it. I could graft all the oled I initialization stuff from fdrs_oled.h into my .ino file but I this k there must be a better way.... ..?
PB

**Comment by aviateur17** [comment]
@NorthBaandee, if you don't want any of the debug information at all you can go into fdrs_oled.h in the drawPageOLED function and on line 114 you can comment out the drawDebugPage(); function call and replace it with a call of whatever information you want to display.  You may also want to change the DISPLAY_PAGES define to 1 to only display a single page you want.     You can define your page in one of the example functions like drawStatusPage() on line 76 and then call that function instead of drawDebugPage() as mentioned earlier.   If you need additional help let us know.

**Reply by NorthBaandee** [reply]
I solved this by cloning fdrs_oled.h and calling it fdrs_nodebug_oled.h and removed all the functions except setup_oled(). Then I added these lines to fdrs_node.h
#ifdef USE_OLED_NODEBUG
 #include fdrs_nodebug_oled.h
#end if
Then I comment out #define USE_OLED
and replace it with #define USE_OLED_NODEBUG¹
and put setup_oled(); in the .ino  file.
I'm making progress now. 
The next problem to solve is why the wifi softAP does not allocate any ip addresses to clients.... ..

**Reply by aviateur17** [reply]
I'm guessing you'd need a DHCP server or to assign them static IP addresses.  Not sure if the SoftAP has built in DHCP capabilities.

**Reply by NorthBaandee** [reply]
I've configured a few soft APs before, they usually work fine with no parameters at all,  defaulting to an IP of 192.168.4.1 and serving anything up to 4 stations. I'll try some different libraries and see what happens. Might be a power issue. I saw some brownout triggers when I try to connect. Might try some 10uF decoupling caps on the 3.3v.... ..

**Reply by NorthBaandee** [reply, marked as answer]
OK solved the problem of why the access point was not accepting clients. I had the gateway IP set to something stupid. If the gateway IP is not 0.0.0.0 the softAP won't accept clients. Who knew?

**Comment by timmbogner** [comment]
It sounds like you really just don't want to use the FDRS OLED function at all. You should be able to disable `USE_OLED` and install/use whichever library you want. I use [Thingpulse's](https://github.com/ThingPulse/esp8266-oled-ssd1306), but Adafruit has also built a competent one.

This sounds a little simpler than your solution, but I'm not sure if I understand. Hope it's working out!

**Comment by NorthBaandee** [comment]
You are correct, it does make this project dependent on FDRS, when I really want FDRS to be an optional feature. The main objective here is a standalone small wturbine controller. Being able to add FDRS as an optional plug-in module would make a great feature on all sorts of machinery.

**Reply by timmbogner** [reply]
I think you may have misunderstood my suggestion. I'm saying that you can disable the FDRS OLED functionality, then implement your own OLED interface using one of the available libraries. I don't think there should be any interference between FDRS and the Adafruit or ThingPulse 1306 OLED libraries if `USE_OLED` is disabled. Write your sketch to collect and display the data on the OLED, then add the required FDRS functions where they are needed. Pretend there is no FDRS OLED functionality and you're doing it all from scratch.

**Comment by NorthBaandee** [comment]
Yep that is what I have done. I have rewritten the display  to use the simple adafruit oled driver and commented out #define USE_OLED in the fdrs_node_config.h file.Also got rid of all this stuff: 
#ifdef USE_OLED_NODEBUG
#include fdrs_nodebug_oled.h
#end if
So now in my .ino file I have 
#define USE_FDRS
and all the loadFDRS(), beginFDRS(), DBG(), sedFDRS functions are bracketed by #ifdef USE_FDRS and #endif.
Seems to work.... ..
